client/src/components/QandA/QuestionsList/QuestionCard/Modal.jsx

```diff
@@ -20,109 +30,181 @@ const Modal = ({ setShowModal, usage, product_name, questionOrProduct_id }) => {
     }
   };
 
+  // Determines displayed text based on whether Q or A
   if (usage === 'addAnswer') {
     title = 'Submit your answer';
     subtitle = `${product_name}:`;
     title_body = 'Your Answer';
+    buttonText = 'Submit Answer';
   } else {
     title = 'Ask Your Question';
     subtitle = `About the ${product_name}:`;
     title_body = 'Your Question';
+    buttonText = 'Submit Question';
   }
 
-  const postModalInput = (endpoint, body) => {
-    // console.log('Modal Post Body: ', body);
-    axios.post(localHost + endpoint, body)
-      .catch(err => console.error('Error posting Modal PUT req: ', err));
+
+  const handleFileInputChange = (e) => {
+    const files = e.target.files;
+    const filesArray = [];
+    for (let i = 0; i < files.length; i++) {
+      filesArray.push(files[i]);
+    }
+
+    setImages(prevImages => [...filesArray, ...prevImages]);
+
+    const previewDIV = document.getElementById('QA-Modal-previews');
+    // while (previewDIV.firstChild) {
+    //   previewDIV.removeChild(previewDIV.lastChild);
+    // }
+
+    filesArray.forEach(imageFile => {
+    const reader = new FileReader();
+    reader.onload = () => {
+      var image = new Image();
+      image.height = 50;
+      image.alt = 'Image to upload';
+      image.src = reader.result;
+      previewDIV.appendChild(image);
+    }
+    reader.readAsDataURL(imageFile);
+    });
   };
 
-  const handleSubmit = (event) => {
+  const tryAgainHandler = (e) => {
+    e.preventDefault();
+    setPreviewSources([]);
+    setImages([]);
+  }
+
+  const handleSubmit = async (event) => {
     event.preventDefault();
+    const imageURLs = [];
+
+    // Uploads images to Cloudinary
+    for (let k = 0; k < images.length; k++) {
+      const data = new FormData();
+      data.append('file', images[k]);
+      data.append('upload_preset', cloudinary_preset);
+      data.append('cloud_name', cloudinary_name);
+
+      try  {
+        const result = await axios.post(cloudinary_url, data);
+        imageURLs.unshift(result.data.url);
+      } catch(err) {
+        console.error( 'Error posting to Cloudinary: ', err );
+      }
+    }
 
     let endpoint;
     let body = {
       body: document.getElementById('QA-modal-body').value,
+      // body: event.target.modalBody.value,
```

Comment:
keeping for refactoring?

Reply:
Possibly but chiefly for learning. Will change by next PR.